app/workers/commit_monitor_handlers/commit/bugzilla_pr_checker.rb

```diff
@@ -0,0 +1,69 @@
+module CommitMonitorHandlers
+  module Commit
+    class BugzillaPrChecker
+      include Sidekiq::Worker
+      include BugzillaCommon
+
+      sidekiq_options :queue => :miq_bot
+
+      def self.handled_branch_modes
+        [:pr]
+      end
+
+      attr_reader :branch, :commit, :message
+
+      def perform(branch_id, commit, commit_details)
+        logger.info("Performing bugzilla PR check on branch #{branch_id}")
+        @branch  = CommitMonitorBranch.where(:id => branch_id).first
+        @commit  = commit
+        @message = commit_details["message"]
+
+        return unless branch_valid?
+
+        MiqToolsServices::Bugzilla.ids_in_git_commit_message(message).each do |bug_id|
+          update_bugzilla_status(bug_id)
+        end
+      end
+
+      private
+
+      def update_bugzilla_status(bug_id)
+        with_bug(bug_id) do |bug|
+          add_pr_comment(bug)
+          update_bug_status(bug)
+          bug.save
```

Comment:
Could you get into a position where save is called but nothing changes?  What happens?